src/macaron/__main__.py

```diff
@@ -62,25 +72,45 @@ def analyze_slsa_levels_single(analyzer_single_args: argparse.Namespace) -> None
 
     run_config = {}
 
-    if analyzer_single_args.repo_path:
-        # Generate a config dict similar to a config read from user yaml file.
+    if analyzer_single_args.config_path:
+        # Get user config from yaml file
+        run_config = YamlLoader.load(analyzer_single_args.config_path)
+        if run_config is None:
+            sys.exit(os.EX_DATAERR)
+    else:
+        repo_path = analyzer_single_args.repo_path
+        purl = analyzer_single_args.package_url
+        branch = analyzer_single_args.branch
+        digest = analyzer_single_args.digest
+
+        if repo_path and purl and not (branch and digest):
+            # To provide the purl together with the repository path, the user must specify the branch and commit
+            # digest.
+            logger.error(
+                "Please provide branch and commit digest for the repo at %s because a PURL %s is also provided",
+                repo_path,
+                purl,
+            )
+            sys.exit(os.EX_USAGE)
+
+        # We need to use empty strings when the input values are of None type. This is because this dictionary will be
+        # passed into the Configuration instance, where the existing values in Configuration.options are replaced by
+        # whatever we assign it here. Technically, the data in ``Configuration`` class are not limited to only strings.
+        # Therefore, it could be cases where the ``purl`` field is initialized as an empty string in the constructor
+        # of the Configuration class, but if `` analyzer_single_args.package_url`` is None, the ``purl`` field is set
+        # to None in the Configuration instance.
+        # This in-consistence could cause potential issues when Macaron handle those inputs.
+        # TODO: improve the implementation of Configuation to remove such non-deterministic behaviors.
```

Comment:
```suggestion
        # This inconsistency could cause potential issues when Macaron handles these inputs.
        # TODO: improve the implementation of `Configuation` class to avoid such inconsistencies.
```

Reply:
Fixed in https://github.com/oracle/macaron/pull/401/commits/f66b0be2bf3c2556eed5885315d39cd66b792306